src/pymor/reductors/basic.py

```diff
@@ -295,6 +297,106 @@ def build_rom(self, projected_operators, error_estimator):
+            # Solve LS problem argmin||Ax - b||_{W} via the normal equation: (A^* W A) x = A^* W b
+            # Note: we assume that A maps to the dual of the test space, therefore due to
+            # Riesz the inverse operator is required to compute the norm in the dual space.
+            W = InverseOperator(self.product) if self.product else None
+
+            projected_operators = {
+                'operator':          project(AdjointOperator(fom.operator, range_product=W) @ fom.operator,
+                                             range_basis=RB, source_basis=RB),
+                'rhs':               project(AdjointOperator(fom.operator, range_product=W) @ fom.rhs,
+                                             range_basis=RB, source_basis=None),
+                'output_functional': project(fom.output_functional, None, RB)
+            }
+
+        else:
+            expanded_op = expand(fom.operator)
+            # See comment in self.use_normal_equations == True: The inverse of self.product is used
+            # in estimate_image if riesz_representative == True.
+            test_space = estimate_image(operators=[expanded_op], domain=RB, orthonormalize=True,
+                                        product=self.product, riesz_representatives=True)
+
+            projected_operators = {
+                'operator':          project(fom.operator, range_basis=test_space,
+                                             source_basis=RB).with_(solver=ScipyLSTSQSolver()),
+                'rhs':               project(fom.rhs, range_basis=test_space, source_basis=None),
+                'products':          {k: project(v, test_space, RB) for k, v in fom.products.items()},
+                'output_functional': project(fom.output_functional, None, RB)
+            }
+        return projected_operators
+
+    def project_operators_to_subbasis(self, dims):
+        rom = self._last_rom
+        dim = dims['RB']
+
+        if self.use_normal_equations:
+            # Square system: both dimensions are the same
+            projected_operators = {
+                'operator':          project_to_subbasis(rom.operator, dim, dim),
+                'rhs':               project_to_subbasis(rom.rhs, dim, None),
+                'products':          {k: project_to_subbasis(v, dim, dim) for k, v in rom.products.items()},
+                'output_functional': project_to_subbasis(rom.output_functional, None, dim)
+            }
+        else:
+            # Rectangular system: test space (range) is larger than trial space (source)
+            expanded_op = expand(self.fom.operator)
+            _, image_dims = estimate_image_hierarchical(operators=[expanded_op],
+                                                        domain=self.RB,
+                                                        orthonormalize=True,
+                                                        product=self.product)
+            range_dims = image_dims[:dim + 1]
```

Comment:
```suggestion
            range_dim = image_dims[dim]
```
or am I missing something?